compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1067,6 +1070,192 @@ class LoweringConfigGenerator {
+      }
+      replaceMap1[tempDimIndex] = totalLoopNum;
+      tempDimIndex++;
+      totalLoopNum++;
+    }
+    applyReplaceMap(replaceMap0);
+    applyReplaceMap(replaceMap1);
+  }
+
+  /// For IndexingMapOpInterface, use its indexing maps for
+  /// propagation.
+  void propagateOnIndexingMapOp(
+      IndexingMapOpInterface indexingMapOp,
+      llvm::EquivalenceClasses<unsigned> &indicesEquivalence,
+      llvm::SmallDenseMap<Value, SmallVector<unsigned>> &valueToGlobalDimMaps) {
+    Operation *op = indexingMapOp.getOperation();
+    for (auto [operand, map] :
+         llvm::zip(indexingMapOp->getOpOperands(),
+                   indexingMapOp.getIndexingMapsArray())) {
+      Value value = operand.get();
+      if (valueToGlobalDimMaps.contains(value)) {
+        for (auto [dim, expr] : llvm::enumerate(map.getResults())) {
+          if (auto dimExpr = dyn_cast<AffineDimExpr>(expr);
+              dimExpr && dim < valueToGlobalDimMaps[value].size()) {
+            unsigned pos = dimExpr.getPosition();
+            indicesEquivalence.unionSets(valueToGlobalDimMaps[value][dim],
+                                         operationToGlobalDimMaps[op][pos]);
+          }
+        }
+      }
+    }
+    auto dsOp = cast<DestinationStyleOpInterface>(op);
+    unsigned numInputs = dsOp.getNumDpsInputs();
+    for (auto [offset, result] : llvm::enumerate(indexingMapOp->getResults())) {
+      AffineMap map = indexingMapOp.getIndexingMapsArray()[numInputs + offset];
+      for (AffineExpr expr : map.getResults()) {
+        if (auto dimExpr = dyn_cast<AffineDimExpr>(expr)) {
+          unsigned pos = dimExpr.getPosition();
+          valueToGlobalDimMaps[result].push_back(
+              operationToGlobalDimMaps[op][pos]);
+        }
+      }
+    }
+  }
+
+  /// For pack and unpack operations, since tiling is applied to the
+  /// outer (unpacked) dimensions, an identity mapping can be applied
+  /// on them.
+  void propagateOnPackUnpackOp(
+      Operation *op, llvm::EquivalenceClasses<unsigned> &indicesEquivalence,
+      llvm::SmallDenseMap<Value, SmallVector<unsigned>> &valueToGlobalDimMaps,
+      unsigned numLoops) {
+
+    for (OpOperand &operand : op->getOpOperands()) {
+      Value value = operand.get();
+      if (valueToGlobalDimMaps.contains(value)) {
+        unsigned rank =
+            static_cast<unsigned>(cast<ShapedType>(value.getType()).getRank());
+        unsigned outDimSize = std::min(rank, numLoops);
+        for (unsigned i = 0; i < outDimSize; ++i) {
+          indicesEquivalence.unionSets(valueToGlobalDimMaps[value][i],
+                                       operationToGlobalDimMaps[op][i]);
+        }
+      }
+    }
+    for (Value result : op->getResults()) {
+      valueToGlobalDimMaps[result] = operationToGlobalDimMaps[op];
+    }
+  }
+
+  /// For other operations, assume an identity
+  /// mapping from any value whose rank matches the operation's loop
+  /// number.
+  void propagateOnDefaultOp(
```

Comment:
ditto:

- Format and refresh the comment a bit.
- Use early-exit in the `valueToGlobaDimMaps.contains(...)`.
- I'd rename it to `propagateOnUnknownOp`, which is clearer than default.